tests/test_core/test_idx_downloader.py

```
@@ -0,0 +1,216 @@
+    for year in sample_config.years:
+        for quarter in sample_config.quarters:
+            url = f"https://www.sec.gov/Archives/edgar/full-index/{year}/QTR{quarter}/master.idx"
+            mock_aioresponse.get(url, status=200, body=f"Content for {year} Q{quarter}")
+    
+    await download_and_process_idx_files(sample_config)
+    
+    # Verify all files were downloaded
+    idx_dir = Path(sample_config.directories.master_idx)
+    for year in sample_config.years:
+        for quarter in sample_config.quarters:
+            idx_file = idx_dir / f"{idx_type}{quarter}{year}.idx"
```

Ruff (0.8.2)

Undefined name `idx_type`

(F821)